src/libraries/Microsoft.Extensions.DependencyInjection/src/Resources/Strings.resx

```diff
@@ -195,4 +195,7 @@
   <data name="KeyedServiceAnyKeyUsedToResolveService" xml:space="preserve">
     <value>KeyedService.AnyKey cannot be used to resolve a single service.</value>
   </data>
+  <data name="NoKeyedServiceRegistered" xml:space="preserve">
+    <value>No keyed service for type '{0}' using key type '{1}' has been registered.</value>
```

Comment:
We don't want to include a ToString()'d key, only its type? Seems like one of the most common types of keys is string, and that string value would be more useful than "string". Is it that we're concerned about leaking details in the error message?

Reply:
We typically don't show disclose user data, but type names are OK.